File: gabe/contains_duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> set;
        
        for(int i = 0; i < nums.size(); i++) {
            if(set.find(nums[i]) == set.end()) {
                set.insert(nums[i]);
            } else {
                return true;
            }
        }
        return false;
        
    }
};

File: greg/day1/contains_duplicate.rs
use std::collections::HashSet;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut set = HashSet::new();
        for i in nums {
            if !set.insert(i) {
                return true;
            }
        }
        return false;
    }
}